fundrebalancer/fundrebalancer.go

```diff
@@ -302,6 +358,54 @@ func (r *FundRebalancer) MoveFundsToChain(
 	return hashes, totalUSDCcMoved, nil
 }
 
+func (r *FundRebalancer) rebalanceID(
```

Comment:
here we are either grabbing a pending transfer between the rebalance to and from chains that already exists, or creating one. this is so that the erc20 approval tx has a rebalance tx id to link back to in the submitted txs table

Reply:
One case where I believe this breaks down is if multiple rebalances are submitted along the route in quick succession. A rebalance transfer could already have been initiated and in the pending state when another rebalance is initiated along the same route. The later would be assigned the same rebalance id right?

This logic feels complex to reason about and somewhat brittle. I would actually be okay if we just didnt associate erc20 approvals with a rebalance transfer. The association is a nice to have and not essential and we might even just do infinite allowances to save gas cost on approvals as you once suggested.

Reply:
got it, i assumed we would want this but i am definitely down to get rid of it since it adds a lot of complexity